doc/source/text.rst

```diff
@@ -201,9 +207,92 @@ and optional groups like
 
 .. ipython:: python
 
-   pd.Series(['a1', 'b2', '3']).str.extract('(?P<letter>[ab])?(?P<digit>\d)')
+   pd.Series(['a1', 'b2', '3']).str.extract('([ab])?(\d)')
+
+can also be used. Note that any capture group names in the regular
+expression will be used for column names; otherwise capture group
+numbers will be used.
+
+Extracting a regular expression with one group returns a DataFrame
+with one column if ``expand=True``, and a Series if ``expand=False``.
+
+.. ipython:: python
+
+   pd.Series(['a1', 'b2', 'c3']).str.extract('[ab](\d)', expand=True)
+   pd.Series(['a1', 'b2', 'c3']).str.extract('[ab](\d)', expand=False)
```

Comment:
since below you are going to show ``expand=True/False``, do these in the same order.

Further I think its ok to break these into separate ``ipython`` blocks (e.g. with your commentary in-between)